apps/rush-lib/src/cli/RushPnpmCommandLine.ts

```diff
@@ -0,0 +1,259 @@
+    ['start', CommandKind.KnownSafe],
+    ['store', CommandKind.KnownSafe],
+    ['test', CommandKind.KnownSafe],
+    [/* synonym */ 't', CommandKind.KnownSafe],
+    ['unlink', CommandKind.ShowWarning],
+    ['update', CommandKind.ShowWarning],
+    [/* synonym */ 'up', CommandKind.ShowWarning]
+  ]);
+
+  public static launch(launcherVersion: string, options: ILaunchRushPnpmInternalOptions): void {
+    // Node.js can sometimes accidentally terminate with a zero exit code  (e.g. for an uncaught
+    // promise exception), so we start with the assumption that the exit code is 1
+    // and set it to 0 only on success.
+    process.exitCode = 1;
+
+    try {
+      // Are we in a Rush repo?
+      let rushConfiguration: RushConfiguration | undefined = undefined;
+      if (RushConfiguration.tryFindRushJsonLocation()) {
+        rushConfiguration = RushConfiguration.loadFromDefaultLocation({ showVerbose: true });
+      }
+
+      NodeJsCompatibility.warnAboutCompatibilityIssues({
+        isRushLib: true,
+        alreadyReportedNodeTooNewError: !!options.alreadyReportedNodeTooNewError,
+        rushConfiguration
+      });
+
+      if (!rushConfiguration) {
+        throw new Error(
+          'The "rush-pnpm" command must be executed in a folder that is under a Rush workspace folder'
+        );
+      }
+
+      if (rushConfiguration.packageManager !== 'pnpm') {
+        throw new Error(
+          'The "rush-pnpm" command requires your rush.json to be configured to use the PNPM package manager'
+        );
+      }
+
+      if (!rushConfiguration.pnpmOptions.useWorkspaces) {
+        throw new Error(
+          'The "rush-pnpm" command requires the "useWorkspaces" setting to be enabled in rush.json'
+        );
+      }
+
+      const workspaceFolder: string = rushConfiguration.commonTempFolder;
+      const workspaceFilePath: string = path.join(workspaceFolder, 'pnpm-workspace.yaml');
+
+      if (!FileSystem.exists(workspaceFilePath)) {
+        console.error(colors.red('Error: The PNPM workspace file has not been generated:'));
+        console.error(`  ${colors.red(workspaceFilePath)}\n`);
+        console.error(colors.cyan(`Do you need to run "rush install" or "rush update"?`));
+        throw new AlreadyReportedError();
+      }
+
+      if (!FileSystem.exists(rushConfiguration.packageManagerToolFilename)) {
+        console.error(colors.red('Error: The PNPM local binary has not been installed yet.'));
+        console.error('\n' + colors.cyan(`Do you need to run "rush install" or "rush update"?`));
+        throw new AlreadyReportedError();
+      }
+
+      // 0 = node.exe
+      // 1 = rush-pnpm
+      const pnpmArgs: string[] = process.argv.slice(2);
+
+      RushPnpmCommandLine._validatePnpmUsage(pnpmArgs);
+
+      const pnpmEnvironmentMap: EnvironmentMap = new EnvironmentMap(process.env);
+      pnpmEnvironmentMap.set('NPM_CONFIG_WORKSPACE_DIR', workspaceFolder);
+
+      if (rushConfiguration.pnpmOptions.pnpmStorePath) {
```

Comment:
We should also populate the `rushConfiguration.pnpmOptions.environmentVariables` here.

Reply:
Also, does Rush itself set any other ENV vars when running `rush install`/`rush update` that may affect how PNPM operates?